azurerm/data_source_arm_public_ips.go

```diff
@@ -0,0 +1,89 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceArmPublicIPs() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmPublicIPsRead,
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+			"minimum_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"attached": {
+				Type:     schema.TypeBool,
+				Required: true,
+				ForceNew: true,
+			},
+			"public_ips": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
+		},
+	}
+}
+
+func dataSourceArmPublicIPsRead(d *schema.ResourceData, meta interface{}) error {
+	publicIPClient := meta.(*ArmClient).publicIPClient
+
+	resGroup := d.Get("resource_group_name").(string)
+	minimumCount, minimumCountOk := d.GetOk("minimum_count")
+	attachedOnly := d.Get("attached").(bool)
+	resp, err := publicIPClient.List(resGroup)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
```

Comment:
can we update this to use the helper method `utils.ResponseWasNotFound(resp)` - which also handles connection drops?